src/Symfony/Component/Finder/Gitignore.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Finder;
+
+/**
+ * Gitignore matches against text.
+ *
+ * @author Ahmed Abdou <mail@ahmd.io>
+ */
+class Gitignore
+{
+    /**
+     * Returns a regexp which is the equivalent of the gitignore pattern.
+     *
+     * @param string $gitIgnoreFileContent
+     *
+     * @return string The regexp
+     */
+    public static function toRegex(string $gitIgnoreFileContent): string
+    {
+        $gitIgnoreFileContent = preg_replace('/^[^\\\\]*#.*/', '', $gitIgnoreFileContent);
+        $gitIgnoreLines = preg_split('/\r\n|\r|\n/', $gitIgnoreFileContent);
+        $gitIgnoreLines = array_map('trim', $gitIgnoreLines);
+        $gitIgnoreLines = array_filter($gitIgnoreLines);
+
+        $ignoreLinesPositive = array_filter($gitIgnoreLines, function (string $line) {
+            return !preg_match('/^!/', $line);
+        });
+
+        $ignoreLinesNegative = array_filter($gitIgnoreLines, function (string $line) {
+            return preg_match('/^!/', $line);
+        });
+
+        $ignoreLinesNegative = array_map(function (string $line) {
+            return preg_replace('/^!(.*)/', '${1}', $line);
+        }, $ignoreLinesNegative);
+        $ignoreLinesNegative = array_map([__CLASS__, 'getRegexFromGitignore'], $ignoreLinesNegative);
+
+        $ignoreLinesPositive = array_map([__CLASS__, 'getRegexFromGitignore'], $ignoreLinesPositive);
+        if (empty($ignoreLinesPositive)) {
+            return '/^$/';
+        }
+
+        if (empty($ignoreLinesNegative)) {
+            return sprintf('/%s/', implode('|', $ignoreLinesPositive));
+        }
+
+        return sprintf('/(?=^(?:(?!(%s)).)*$)(%s)/', implode('|', $ignoreLinesNegative), implode('|', $ignoreLinesPositive));
+    }
+
+    private static function getRegexFromGitignore(string $gitignorePattern): string
```

Comment:
`$gitignorePattern` vs. `$gitIgnoreFileContent` :D (i would repeat it in the variable names)

Reply:
@ro0NL 
loooooool               
I completely missed this :D                  
all fixed now
please check and tell me if all is okay now